### PR TITLE
feat: support classic-flavored ingest keys

### PR DESCRIPTION
**Pull request by robbkidd**
## Which problem is this PR solving?

There's a new key format on the block: ingest keys. To determine the correct dataset defaults for classic environments, the key detection logic needs an update. So that the details of the logic are implemented in fewer places, expose the logic as public methods, primarily for Beeline use. 

## Short description of the changes

- update classic key detection logic
- promote `Libhoney::Client.classic_write_key?` to public instance method
- also provide same logic as a module-level method `Libhoney.classic_write_key?` so the check can be made without an instance of the client

## To Consider

- Is the dated "write_key" vocabulary something we want to keep as this method goes public? Maybe provide aliased `classic_api_key?` methods as well?

### CODE REVIEW

File: test/libhoney_test.rb [hunk 1]
```diff
@@ -53,6 +53,18 @@ def test_initialize_with_params
     assert_equal 1500, honey.pending_work_capacity
   end
 
+  def test_initialize_with_classic_v3_ingestkey_and_dataset
+    #                                       1         2         3         4         5
+    classic_v3_ingest_key = 'hcaic_1234567890123456789012345678901234567890123456789012345678'
```
Comment (robbkidd) [inline test/libhoney_test.rb:58]: Or I could `SecureRandom.hex(58/2)` to get a 58-character string to add to the prefix. ¯\\\_(ツ)\_/¯

**Comment by jharley** [comment]
> Is the dated "write_key" vocabulary something we want to keep as this method goes public?

This is a very thoughtful question, but, I think in libhoney land 'write key' is a deeply engrained so maybe not worth the modernization? Though, no harm in adding it :)

**Review by jharley — APPROVED** [review]
Looks great ✨

**Review by MikeGoldsmith — CHANGES_REQUESTED** [review]
I would prefer to see api_key replace write_key if we're making things public. It is an older term that we do not use outside of libhoney's. This isn't a function we anticipate external users to interact anyway with so happy to move towards conformity and reduce maintenance confusion.

File: lib/libhoney/client.rb [hunk 2]
```diff
@@ -293,7 +293,7 @@ def get_dataset(dataset, write_key)
     end
 
     def classic_write_key?(write_key)
```
Comment (MikeGoldsmith) [inline lib/libhoney/client.rb:295]: Should this func be updated to `claassic_api_key` too?

**Review by MikeGoldsmith — APPROVED** [review]

File: lib/libhoney/client.rb (hunk 2 above)
Comment (robbkidd) [reply lib/libhoney/client.rb:295]: It's not obvious from the diff, but this method is private. I left it named as is to minimize changes, but I'm happy to rename it, too, is service of consistency.

Comment (MikeGoldsmith) [reply lib/libhoney/client.rb:295]: Yeah, let's go for consistency. It's a simple change but we'll likely forget to do it later 😄

Comment (robbkidd) [reply lib/libhoney/client.rb]: Done in 2a1a1fe!

**Pull request merged**